File: cjeongmin/N과_M_(11).js
const fs = require('fs');
const input = fs.readFileSync(0).toString().trim().split('\n');

const [N, M] = input[0].split(' ').map(Number);
const arr = input[1].split(' ').map(Number);

const unique = [...new Set(arr)].sort((a, b) => a - b);
const result = [];

const dfs = (comb) => {
  if (comb.length === M) {
    result.push(comb.join(' '));
    return;
  }

  for (let i = 0; i < unique.length; i++) {
    comb.push(unique[i]);
    dfs(comb);
    comb.pop();
  }
};

dfs([]);
console.log(result.join('\n'));

File: jiho/N과 M (11).py
[n, m] = list(map(int, input().split()))
nums = sorted([x for x in set(list(map(int, input().split())))])


def solution(stack):
    global m, nums

    if len(stack) == m:
        print(" ".join(map(str, stack)))
        return

    for num in nums:
        stack.append(num)
        solution(stack)
        stack.pop()


solution([])
